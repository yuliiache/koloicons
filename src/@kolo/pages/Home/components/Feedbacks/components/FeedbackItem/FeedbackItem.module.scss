@import "@kolo/design";

.feedbackItem {
  margin: 0 16px;
  display: flex;
  flex-direction: column;
  width: calc(33.3% - 32px);
  text-align: center;

  @media screen and (max-width: $mobileWidth) {
    width: 100%;
  }

  img {
    display: block;
    align-self: center;
    width: 56px;
    height: 56px;
    margin-bottom: 16px;
    border-radius: 50%;
  }
}

.feedbackText {
  margin-top: 0;
  font-style: italic;
  font-size: 0.7rem;
  line-height: 1.4;
  color: $secFontColor;
}

.feedbackInfo {
  font-size: 0.6rem;
  line-height: 1.3;
  color: $GreyColor;
}
