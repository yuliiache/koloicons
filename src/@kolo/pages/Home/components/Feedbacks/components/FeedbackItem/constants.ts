export const FEEDBACK_TEXT_TEST_ID = 'feedbackText';
