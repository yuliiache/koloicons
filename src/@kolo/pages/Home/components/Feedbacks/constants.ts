export const FEEDBACKS_KEY = 'feedbacks';
