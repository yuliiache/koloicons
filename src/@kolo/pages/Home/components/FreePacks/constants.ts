export const FREE_PACKS_KEY = 'freePacks';
export const FREE_PACKS_TEST_ID = 'packItemComponent';
