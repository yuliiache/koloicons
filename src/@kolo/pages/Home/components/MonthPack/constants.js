export const MONTH_PACK_KEY = 'monthPack';
