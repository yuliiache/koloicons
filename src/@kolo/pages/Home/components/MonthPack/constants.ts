export const MONTH_PACK_KEY = 'monthPack';
export const MONTH_PACK_TEST_ID = 'monthPackComponent';
