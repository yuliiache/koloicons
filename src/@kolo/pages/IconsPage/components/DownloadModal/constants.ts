export const ICON_KEY = 'icon';
