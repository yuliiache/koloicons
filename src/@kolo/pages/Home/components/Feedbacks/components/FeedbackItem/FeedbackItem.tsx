import {Feedback} from '@kolo/facade/getFeedbacks/types';
import {IMG_URL} from '@kolo/services/helpers/constants';
import {Image} from '@kolo/uiKit/Image/Image';
import Link from '@kolo/uiKit/Link/Link';
import {FC} from 'react';

import {FEEDBACK_TEXT_TEST_ID} from './constants';
import styles from './FeedbackItem.module.scss';

const FeedbackItem: FC<Feedback> = ({name, text, company, avatarPath, url}) => {
  return (
    <div className={styles.feedbackItem}>
      <Image
        src={`${IMG_URL}/${avatarPath}`}
        width={56}
        height={56}
        alt={name}
      />
      <p
        className={styles.feedbackText}
        data-testid={FEEDBACK_TEXT_TEST_ID}
      >
        {text}
      </p>
      <span>
        <Link url={url}>{`${name}, ${company}`}</Link>
      </span>
    </div>
  );
};

export default FeedbackItem;
