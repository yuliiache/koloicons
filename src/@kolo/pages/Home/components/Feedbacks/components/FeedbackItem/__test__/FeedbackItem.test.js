import React from 'react';
import {render} from '@testing-library/react';

import {FEEDBACK_TEXT_TEST_ID} from '../constants';
import FeedbackItem from '../FeedbackItem';

const commonProps = {
  name: 'John Doe',
  text: 'This is a sample feedback text.',
  company: 'Sample Company',
  avatarPath: 'avatar.jpg',
  url: 'https://example.com',
};

describe('FeedbackItem Component', () => {
  it('renders snapshot correctly', () => {
    const {asFragment} = render(<FeedbackItem {...commonProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders text correctly', () => {
    const text = 'Sample text';
    const {getByTestId} = render(
      <FeedbackItem
        {...commonProps}
        text={text}
      />
    );

    const feedbackTextElement = getByTestId(FEEDBACK_TEXT_TEST_ID);

    expect(feedbackTextElement.textContent).toBe(text);
  });
});
