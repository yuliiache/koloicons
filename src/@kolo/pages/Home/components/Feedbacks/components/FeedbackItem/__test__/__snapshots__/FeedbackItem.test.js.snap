// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackItem Component renders snapshot correctly 1`] = `
<DocumentFragment>
  <div
    class="feedbackItem"
  >
    <img
      alt="John Doe"
      height="56"
      src="undefined/avatar.jpg"
      width="56"
    />
    <p
      class="feedbackText"
      data-testid="feedbackText"
    >
      This is a sample feedback text.
    </p>
    <span>
      <a
        class="link link--default"
        data-testid="externalLinkComponent"
        href="https://example.com"
        rel="noopener noreferrer nofollow"
      >
        John Doe, Sample Company
      </a>
    </span>
  </div>
</DocumentFragment>
`;
